src/Expect.elm

```diff
@@ -647,3 +637,53 @@ allHelp list query =
 
                 outcome ->
                     outcome
+
+
+
+{---- Private helper functions ----}
+
+
+reportFailure : String -> String -> String -> Expectation
+reportFailure comparison expected actual =
+    Test.Expectation.Fail
+        { given = ""
+        , description = comparison
+        , reason = Test.Expectation.Comparison (toString expected) (toString actual)
+        }
+
+
+reportCollectionFailure : String -> a -> b -> List c -> List d -> Expectation
+reportCollectionFailure comparison expected actual missingKeys extraKeys =
+    Test.Expectation.Fail
+        { given = ""
+        , description = comparison
+        , reason =
+            Test.Expectation.CollectionDiff
+                { expected = toString expected
+                , actual = toString actual
+                , extra = List.map toString extraKeys
+                , missing = List.map toString missingKeys
+                }
```

Comment:
🎨 with blocks like this, I generally find piping the record into the constructor easier to read. The rule should be: indent as little as possible.